### PR TITLE
#217 change delay

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**

### CODE REVIEW

File: gui/doc/input.csv [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ﻿Name,Type,Default,Doc
 *value,dynamic(any),None,The value represented by this control.
+change_delay,int,<i>App config</i>,"minimum time between triggering two <i>on_change</i> callback.
```
Comment (FabienLelaquais) [inline gui/doc/input.csv]: Capital 'Minimum...'
"two <i>on_change</i> calls"

File: gui/doc/input.csv [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
+The default value is defined at the application configuration level by change_delay. if None or 0, the delay is set to 300ms."
```
Comment (FabienLelaquais) [inline gui/doc/input.csv]: "configuration level by change_delay"
->
<pre>
configuration level by the **change_delay** configuration option
</pre>

I guess that if this was set to 0, I would want no delay at all... Not a (not so) strange default value, that makes a lot of sense if None.

File: gui/doc/slider.csv [hunk 3]
```diff
@@ -17,10 +17,15 @@ Possible values are:
 labels,bool|dict,,"The labels for specific points of the slider.
 If set to <i>True</i>, this slider uses the labels of the <i>lov</i> if there are any.
 If set to a dictionary, the slider uses the dictionary keys as a <i>lov</i> key or index, and the associated value as the label."
+continuous,bool,True,"If set to <i>False</i>, the control emits an on_change notification only when the cursor is released,
```
Comment (FabienLelaquais) [inline gui/doc/slider.csv]: <pre><i>False</i></pre> => "False" (no italics)
on_change -> <pre>_on_change_</pre>

"cursor" -> "mouse button"

File: gui/doc/slider.csv [hunk 4]
```diff
@@ -17,10 +17,15 @@ Possible values are:
 labels,bool|dict,,"The labels for specific points of the slider.
 If set to <i>True</i>, this slider uses the labels of the <i>lov</i> if there are any.
 If set to a dictionary, the slider uses the dictionary keys as a <i>lov</i> key or index, and the associated value as the label."
+continuous,bool,True,"If set to <i>False</i>, the control emits an on_change notification only when the cursor is released, 
+otherwise notifications are emitted during the cursor movements. 
+If <i>lov</i> is defined, the default value is <i>False</i>."
```
Comment (FabienLelaquais) [inline gui/doc/slider.csv]: <pre><i>False</i></pre> => "False" (no italics)

File: gui/doc/slider.csv [hunk 5]
```diff
@@ -17,10 +17,15 @@ Possible values are:
...
+change_delay,int,<i>App config</i>,"minimum time between triggering two <i>on_change</i> callback.
```
Comment (FabienLelaquais) [inline gui/doc/slider.csv]: Capital 'Minimum...'
"two <i>on_change</i> calls"

File: gui/src/components/Taipy/Input.tsx [hunk 6]
```diff
@@ -13,6 +13,7 @@ const Input = (props: TaipyInputProps) => {
     const { dispatch } = useContext(TaipyContext);
     const delayCall = useRef(-1);
 
+    const changeDelay = (typeof props.changeDelay === "number" && props.changeDelay > 0) ? props.changeDelay : 300;
```
Comment (FabienLelaquais) [inline gui/src/components/Taipy/Input.tsx]: const changeDelay = (typeof props.changeDelay === "number" && props.changeDelay >= 0) ? props.changeDelay : 300;

File: gui/doc/slider.csv [hunk 7]
```diff
@@ -17,10 +17,15 @@ Possible values are:
...
+The default value is defined at the application configuration level by change_delay. if None or 0, there's no delay."
```
Comment (FabienLelaquais) [inline gui/doc/slider.csv]: "configuration level by change_delay"
->
<pre>
configuration level by the **change_delay** configuration option
</pre>

File: gui/doc/tree.csv [hunk 8]
```diff
@@ -1,3 +1,4 @@
 ﻿Name,Type,Default,Doc
 expanded,dynamic(bool|str[]),True,"if boolean value False: only one node can be expanded at one given level else it should be an array of the ids that need to be expanded"
+multiple,bool,FALSE,"If <i>True</i>, the user can select multiple items by holding the Ctrl key while clicking on items."
```
Comment (FabienLelaquais) [inline gui/doc/tree.csv]: <pre><i>True</i></pre> -> True (no italics)

**Review by FabienLelaquais — APPROVED** [review]
Minor doc changes - and one more semantic...

File: gui/doc/input.csv (hunk 2 above)
Comment (FredLL-Avaiga) [reply gui/doc/input.csv]: is markdown supported in csv ?

**Pull request merged**